x-pack/legacy/plugins/apm/server/lib/index_pattern/create_static_index_pattern.test.ts

```diff
@@ -4,14 +4,23 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import { Server } from 'hapi';
 import { createStaticIndexPattern } from './create_static_index_pattern';
 import { Setup } from '../helpers/setup_request';
 import * as savedObjectsClient from '../helpers/saved_objects_client';
 import * as HistoricalAgentData from '../services/get_services/has_historical_agent_data';
+import { APMRequestHandlerContext } from '../../routes/typings';
 
-function getMockConfig(config: Record<string, unknown>) {
-  return () => ({ get: (key: string) => config[key] });
+function getMockContext(config: Record<string, unknown>) {
+  return ({
+    config,
+    __LEGACY: {
+      server: {
+        savedObjects: {
+          getSavedObjectsRepository: jest.fn()
+        }
+      }
+    }
+  } as unknown) as APMRequestHandlerContext;
```

Comment:
What do you think about creating the saved objects client on `Setup` like we do with `client` and `internalClient`. then we could have `savedObjectsClient` and `internalSavedObjectsClient`. This way we wouldn't have to pass context around anymore.

Reply:
`context` is still needed for `config` and some other stuff, or am I misunderstanding? the internal saved objects client is a WIP, maybe better to wait? https://github.com/elastic/kibana/pull/48882

Reply:
`config` is already on `Setup`. I'm thinking that it would be nice if downstream handlers don't have to know about `Context` or `Request`  - at least if they are just doing normal stuff like reading config, calling (internal) savedobjects or (internal) esClient.

Reply:
Right, I see what you mean. Ok if I create an issue for this that we can pick up after the migration to NP is complete (including the usage collector and the internal saved objects client)?

Reply:
Went ahead and created the issue: https://github.com/elastic/kibana/issues/51257